cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -523,6 +578,20 @@ func (o *objectGraph) setSoftOwnership() {
 			}
 		}
 	}
+
+	clusterClasses := o.getClusterClasses()
+	// ClusterClass that are used by atleast one cluster are soft owned by that cluster.
+	for _, clusterClass := range clusterClasses {
+		for _, cluster := range clusters {
+			// if the cluster uses a managed topoloy and uses the clusterclass
+			// set the cluster as a soft owner of the clusterclass.
+			if className, ok := cluster.additionalInfo[clusterTopologyNameKey]; ok {
+				if className == clusterClass.identity.Name && clusterClass.identity.Namespace == cluster.identity.Namespace {
```

Comment:
Let's check also group

Reply:
Just for my understanding: why would we want to check the group?

Here we are making sure that the clusterclass used in the cluster spec (`cluster.spec.topology.class`) matches the clusterclass we are checking against and we are making sure that the cluster and the clusterclass belong to the same namespace because we know that this is a requirement. 
By checking groups we are saying that clusters and clusterclasses should belong to the same k8s group `cluster.x-k8s.io`. I dont see how this is a requirement to establish a soft ownership.

Am I missing something?

Reply:
No you are right.
I did not look at the code above and I did assume we were looping through all the objects, not only clusters